server.js

```diff
@@ -53,51 +53,31 @@ app.post('/api/user', (req, res) => {
     // It is good practice to specifically pick the fields we want to insert here *in the backend*,
     // even if we have already done so on the front end. This is to prevent malicious users
     // from adding unexpected fields by modifying the front end JS in the browser.
-
     var newUser =  _.pick(req.body, [
         'name', 'email', 'phone', 'passphrase'])
-    newUser.isAdmin = false
-    newUser.approvedBy = ''
-    bcrypt.hash(newUser.passphrase, 10, (err, hash) => {
-        // Store hash in database
-        newUser.passphrase = hash
-        db.insertOne('user', newUser).then(result => {
-            return res.json(result)
-        }).catch(error => {
-            console.log(error)
-            return res.status(422).json(error)
-        })
-    })
-})
-
-app.post('/api/posts', function(req, res) {
-    delete req.body['_id']
-    db.insertOne(posts_db_name, req.body).then(response => {
-        res.status(201).json(response);
-    }).catch(error => {
-        res.status(406).json({'error': error})
-    })
-})
-
-
-app.put('/api/posts', function(req, res) {
-    db.updateOneById(posts_db_name, req.body).then(response => {
-        res.status(201).json({msg: 'successfully edited report'});
+    createNewUser(newUser).then(result => {
+        return res.json(result)
     }).catch(error => {
-        console.log('error is', error)
-        res.status(406).json({'error': error})
+        console.log(error)
+        return res.status(422).json(error)
     })
 })
 
-app.get('/api/posts/delete/:id', function(req, res) {
-    const postId = req.params.id
-    db.deleteOne(posts_db_name, postId).then(response => {
-        res.status(201).json({msg: 'successfully deleted disaster report'});
-    }).catch(error => {
-        console.log('error is', error)
-        res.status(406).json({'error': error})
+const createNewUser = (newUser) => {
+    return new Promise( (resolve, reject) => {
```

Comment:
Same thing here - best not to mix callbacks and promises.
```js
const createNewUser = (newUser) => {
    return bcrypt.hash(newUser.passphrase, 10).then(hash => {
        // Safer to use `Object.assign` here.
        // Directly modifying `newUser` can potentially lead to a memory leak.
        return db.insertOne('users', Object.assign({passphrase: hash}, newUser))
    })
}
```

Reply:
+1